src/braket/aws/aws_quantum_task.py

```diff
@@ -426,6 +426,12 @@ def _format_result(result):
 
 @_format_result.register
 def _(result: GateModelTaskResult) -> GateModelQuantumTaskResult:
+    if result.resultTypes:
```

Comment:
There's also other conversions in here: https://github.com/aws/amazon-braket-sdk-python/blob/main/src/braket/tasks/gate_model_quantum_task_result.py#L220. These checks should probably get migrated into a function, that both from_string and this function call

Reply:
Updated